handler/push.go

```diff
@@ -265,7 +265,12 @@ func splitLabels(labels string) (map[string]string, error) {
 	if len(components)%2 != 0 {
 		return nil, fmt.Errorf("odd number of components in label string %q", labels)
 	}
+
 	for i := 0; i < len(components)-1; i += 2 {
+		if !model.LabelNameRE.MatchString(components[i]) ||
+			strings.HasPrefix(components[i], model.ReservedLabelPrefix) {
+			return nil, fmt.Errorf("improper label name %s", components[i])
```

Comment:
Nitpick: Better to use `%q` here (as [a few lines above](https://github.com/prometheus/pushgateway/pull/64/files#diff-afffdd5e76af154b0c3d69694a63a918R266)).